src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core'
+import { Segment } from '../../../core/models/segment'
+import { ConfigurationService } from '../../../shared/services/configuration-service'
+import { StudioConfiguration } from '../../../shared/services/studio-configuration'
+import { Tv2VideoClipAction } from '../../../shared/models/tv2-action'
+import { Subscription } from 'rxjs'
+import { ActionService } from '../../../shared/abstractions/action.service'
+import { MediaStateService } from '../../../shared/services/media-state.service'
+import { Media } from '../../../shared/services/media'
+
+@Component({
+  selector: 'sofie-mini-shelf',
+  styleUrls: ['./mini-shelf.component.scss'],
+  templateUrl: './mini-shelf.component.html',
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
+  protected mediaDuration: number = 0
+  private configurationMediaPreviewUrl: string
+  private configurationServiceSubscription: Subscription
+  private executeActionSubscription: Subscription
+  private serverPostrollDuration: number = 0
```

Comment:
The `configurationMediaPreviewUrl` and `serverPostrollDuration` are both attributes on the same interface `StudioConfiguration`.
Instead of saving all the attributes individually you should just store the `StudioConfiguration` which you are fetching anyway.

Utilize the objects :D